table_schema_cache.go

```diff
@@ -293,6 +293,31 @@ func (c TableSchemaCache) Get(database, table string) *TableSchema {
 	return c[fullTableName(database, table)]
 }
 
+// Helper to sort a given map of tables with a second list giving a priority.
+// If an element is present in the input and the priority lists, the item will
+// appear first (in the order of the priority list), all other items appear in
+// the order given in the input
+func (c TableSchemaCache) GetTableListWithPriority(priorityList []string) (tableNames []string) {
+	// just a fast lookup if the list contains items already
+	contains := map[string]bool{}
+	if len(priorityList) >= 0 {
+		for _, tableName := range priorityList {
+			// ignore tables given in the priority list that we don't know
+			if c[tableName] != nil {
```

Comment:
`if _, found := c[tableName]; found {`

Otherwise the code might panic?

Reply:
happy to change, but in what situation will it panic?

from the golang language spec:

https://golang.org/doc/effective_go.html?h=An+attempt+to+fetch+a+map+value+with+a+key+that

> An attempt to fetch a map value with a key that is not present in the map will return the zero value for the type of the entries in the map. For instance, if the map contains integers, looking up a non-existent key will return 0. A set can be implemented as a map with value type bool. Set the map entry to true to put the value in the set, and then test it by simple indexing.

thus, we will receive a `nil` , which is the null-value for the pointer.

Not arguing against your suggestion - it's "the go way" and I'll definitely improve it, but want to make sure I'm not misunderstanding why you're suggesting it, as I don't see a path leading to panic